cg/apps/demultiplex/flowcell.py

```diff
@@ -0,0 +1,5083 @@
+import logging
+from pathlib import Path
+
+from cg.models.demultiplex.run_parameters import RunParameters
+from cgmodels.demultiplex.sample_sheet import get_sample_sheet_from_file
+from cgmodels.exceptions import SampleSheetError
+from pydantic import ValidationError
+
+LOG = logging.getLogger(__name__)
+
+
+class Flowcell:
+    """Class to collect information about flowcell directories and there particular files"""
+
+    def __init__(self, flowcell_path: Path):
+        LOG.info("Instantiating Flowcell with path %s", flowcell_path)
+        self.path = flowcell_path
+        LOG.info("Set flowcell id to %s", self.flowcell_id)
+
+    @property
+    def flowcell_id(self) -> str:
```

Comment:
This is executed every time it's called. Its not a big deal computationally but you could create the class object with property self._flowcell_id = None. The first time self.flowcell_id is used and field is None, the method is called and property defined. Otherwise the value of self._flowcell_id is returned. Does not matter much now because its a simple path operation. But can matter in some other situations!

Reply:
Yes I agree! I thought about this while implementing and I think you are right